packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2396,19 +2417,20 @@ function commitMutationEffects(
 ) {
   let fiber = firstChild;
   while (fiber !== null) {
-    if (fiber.deletions !== null) {
-      commitMutationEffectsDeletions(
-        fiber.deletions,
-        root,
-        renderPriorityLevel,
-      );
+    const deletions = fiber.deletions;
+    if (deletions !== null) {
+      commitMutationEffectsDeletions(deletions, root, renderPriorityLevel);
 
-      // TODO (effects) Don't clear this yet; we may need to cleanup passive effects
-      fiber.deletions = null;
+      // If there are no pending passive effects, clear the deletions Array.
```

Comment:
Can move this to passive phase, too

Reply:
Hm, true. Same concern as previously expressed about the number of subtle changes (many non-observable) hinge on the fact that we removed ab optimization though.